ts/src/xt.ts

```diff
@@ -0,0 +1,4094 @@
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        return await this.fetchOrdersByStatus ('closed', symbol, since, limit, params);
+    }
+
+    async fetchCanceledOrders (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchCanceledOrders
+         * @description fetches information on multiple canceled orders made by the user
+         * @see https://doc.xt.com/#orderhistoryOrderGet
+         * @see https://doc.xt.com/#futures_ordergetOrders
+         * @param {string|undefined} symbol unified market symbol of the market the orders were made in
+         * @param {int|undefined} since timestamp in ms of the earliest order
+         * @param {int|undefined} limit the maximum number of order structures to retrieve
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        return await this.fetchOrdersByStatus ('canceled', symbol, since, limit, params);
+    }
+
+    async cancelOrder (id: string, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#cancelOrder
+         * @description cancels an open order
+         * @see https://doc.xt.com/#orderorderDel
+         * @see https://doc.xt.com/#futures_ordercancel
+         * @param {string} id order id
+         * @param {string|undefined} symbol unified symbol of the market the order was made in
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const request = {
+            'orderId': id,
+        };
+        let type = undefined;
+        let subType = undefined;
+        let response = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('cancelOrder', market, params);
+        [ subType, params ] = this.handleSubTypeAndParams ('cancelOrder', market, params);
+        if (subType === 'inverse') {
+            response = await this.privateInversePostFutureTradeV1OrderCancel (this.extend (request, params));
+        } else if ((subType === 'linear') || (type === 'swap') || (type === 'future')) {
+            response = await this.privateLinearPostFutureTradeV1OrderCancel (this.extend (request, params));
+        } else {
+            response = await this.privateSpotDeleteOrderOrderId (this.extend (request, params));
```

Comment:
@Dan-krm hum cancelOrder is not working apparently?
```
n xt cancelOrder "'214589946237571584'" 
2023-04-07T10:44:41.335Z
Node.js: v18.14.0
CCXT v3.0.51
xt.cancelOrder (214589946237571584)
ExchangeError xt {"rc":1,"mc":"订单不存在","ma":[],"result":null}
---------------------------------------------------
[ExchangeError] xt {"rc":1,"mc":"订单不存在","ma":[],"result":null}

    at handleErrors               Users/cjg/Git/ccxt2/ccxt/js/src/xt.js:4003             
    at                            Users/cjg/Git/ccxt2/ccxt/js/src/base/Exchange.js:748   
    at processTicksAndRejections  node:internal/process/task_queues:95                   
    at fetch2                     Users/cjg/Git/ccxt2/ccxt/js/src/base/Exchange.js:2430  
    at request                    Users/cjg/Git/ccxt2/ccxt/js/src/base/Exchange.js:2433  
    at cancelOrder                Users/cjg/Git/ccxt2/ccxt/js/src/xt.js:2595             
    at async run                  Users/cjg/Git/ccxt2/ccxt/examples/js/cli.js:295        

xt {"rc":1,"mc":"订单不存在","ma":[],"result":null}
```

Reply:
cancelOrder requires an integer for the id argument, let me know how you think I should handle this, parseInt doesn't work in certain cases because it's rounding the last number:
```
xt.cancelOrder (214922773985805824)
fetch Request:
 xt DELETE https://sapi.xt.com/v4/order/214922773985805820
```

Reply:
fetchOrder requires an integer like this as well, let me know how I should handle this

Reply:
@Dan-krm humm, if it is rounding it, means it does not fit in a regular integer 🤔